### DISCUSSION TITLE
Using swap with theme-change

### DISCUSSION THREAD

**Discussion by zed-wong** [saadeghi/daisyui#1427] [Q&A] [answered]
In [theme-change](https://github.com/saadeghi/theme-change#using-buttons-to-set-a-theme), `set-theme` is inside a button.
`<button data-set-theme="" data-act-class="ACTIVECLASS"></button>`

But in [swap](https://daisyui.com/components/swap/), the swap icon is inside a label
`<label class="swap swap-rotate"></label>`

I've tried to use the swap inside a `button` but the icon didn't swap, and if I use it inside a label the theme won't be changed, what's the best way to do that?

**Comment by zed-wong** [comment, marked as answer]
Never mind, I found an answer here, https://github.com/saadeghi/daisyui/discussions/808

**Comment by zed-wong** [comment]
@saadeghi By the way, is it possible to change SVG icons to the dark theme? Or do I have to make a dark version for each of the icons?

**Reply by saadeghi** [reply]
Ideally, you would want to put the SVG tag inside your HTML because this way you can control everything using Tailwind CSS classes (such as fill color or stroke color)

But if your SVG is a file and you're using it with a `<img>` tag, one way would be inverting the color for a dark theme like this:
```html
<img class="w-10 h-10 [[data-theme=dark]_&]:invert" src="https://www.svgrepo.com/show/6977/star.svg">
```
`[[data-theme=dark]_&]:invert` class inverts the svg colors only on `dark` theme  
Example: https://play.tailwindcss.com/ph25d8pH0j

**Reply by zed-wong** [reply]
Got it, thank you so much!

**Comment by zed-wong** [comment]
@saadeghi Sorry to ask another dumb question, here is my style.css, how can I apply dark mode to the background-image? Thanks!
```
body {
	min-height: 100vh;
	margin: 0;
	background-attachment: fixed;
	background-color: var(--color-bg-0);
	background-size: 100vw 100vh;
	background-image: radial-gradient(
			50% 50% at 50% 50%,
			rgba(255, 255, 255, 0.75) 0%,
			rgba(255, 255, 255, 0) 100%
		),
		linear-gradient(180deg, var(--color-bg-0) 0%, var(--color-bg-1) 15%, var(--color-bg-2) 50%);
}
```

**Reply by saadeghi** [reply]
Nothing to be sorry about. I would be glad to answer questions. But from now on please open new discussion for new questions so it would be easier for other people to find answers here.

Here's the code:
```css
[data-theme=dark] body{
  /* any style here works only for `dark` theme */
}
```
